### ISSUE TITLE
Provide annotation data outside of the data-join

### ISSUE THREAD

**Issue by ColinEberhardt** [d3fc/d3fc#1157]
_From @ColinEberhardt on February 21, 2017 9:20_

If you want to add a single line annotation to a chart, at a fixed location, currently this involves merging the annotation data with your series data and adding a multi-series with a mapping.

A lot of hassle!

I#d like to see something like this:

```
var data = // series data, as an array

var annotation = fc.annotationSvgLine()
  .data([23, 45]);

var series = fc.seriesSvgLine();

var multi = fc.seriesSvgMulti()
  .series([annotation, series]);

d3.select('chart')
  .datum(data)
 .call(multi)
```

This removes the need to merge the annotation data with the series data and provide a multi-mapping.

_Copied from original issue: d3fc/d3fc-annotation#28_

**Comment by chrisprice** [comment]
I agree that's a cleaner API but it does feel like it breaks the pattern and could boil down to some awkward edge cases. Maybe a compromise could be a component which acts to freeze the data bound to the wrapped component e.g. -

```js
var data = // series data, as an array

var annotation = fc.annotationSvgLine();

var series = fc.seriesSvgLine();

var multi = fc.seriesSvgMulti()
  .series([
    fc.bind(annotation, [23, 45]),
    series
  ]);

d3.select('chart')
  .datum(data)
 .call(multi)
```

**Comment by ColinEberhardt** [comment]
I presume `fc.bind` could be use to bind data to any D3 / d3fc component? If so ... I like that idea! I can see other of uses for it, e.g. adding trendlines to charts
